lib/runtime/storage/storagediff.go

```diff
@@ -141,13 +141,17 @@ func (cs *storageDiff) clearPrefixInChild(keyToChild string, prefix []byte,
 // optional limit. It returns the number of keys deleted and a boolean
 // indicating if all keys with the prefix were removed.
 func (cs *storageDiff) clearPrefix(prefix []byte, trieKeys []string, limit int) (deleted uint32, allDeleted bool) {
-	allKeys := slices.Clone(trieKeys)
 	newKeys := maps.Keys(cs.upserts)
-	allKeys = append(allKeys, newKeys...)
+	keysToClear := maps.Keys(cs.upserts)
+	for _, k := range trieKeys {
+		if _, ok := cs.upserts[k]; !ok {
+			keysToClear = append(keysToClear, k)
+		}
+	}
 
 	deleted = 0
-	sort.Strings(allKeys)
-	for _, k := range allKeys {
+	sort.Strings(keysToClear)
```

Comment:
why do we have to sort?

Reply:
It seems to break the correctness of tests without sorting. It looks like a correctness issue to me, though maybe it is just how the tests our set up. It seems key deletion happens on sorted keys, so maybe it is for consistency here? Will ping @EclesioMeloJunior and @dimartiro to see if they have any insight into why the sorting logic needs to occur in our design

Reply:
We are sorting them because we need to delete the keys lexicographically